### DISCUSSION TITLE
Laravel Socialite: Getting Laravel  `\Socialite\Two\InvalidStateException` while using it with Twitter/X OAuth2.0

### DISCUSSION THREAD

**Discussion by MUHAMMADSiyab** [laravel/framework#50360] [Q&A]
Building a laravel app that has let the users to login using their Twitter/X account. After authorization, when redirected to the callback url, getting InvalidStateException. Tried a ton of methods including using ngrok (to have https) but nothing worked. Any help would highly be appreciated.

Using:

- **Linux Mint 21.2 Cinnamon**
- **PHP 8.1.2**
- **Laravel 10.42.0**
- **Socialite ^5.12**

PHP

```
public function twitter()
{
   return Socialite::driver('twitter-oauth-2')->redirect();
}

public function twitterRedirect()
{ 

   $user = Socialite::driver('twitter-oauth-2')->user();

   dd($user);
}
```

**Comment by BiancaAsavoaei** [comment]
Hi! I have the same problem. Did you solve it?

**Reply by MUHAMMADSiyab** [reply]
Not yet. It seems like a bug in the latest version.

**Reply by tesguru** [reply]
> Not yet. It seems like a bug in the latest version.

Try using laragon to create virtual host and https

**Comment by gabefgonc** [comment]
hello, having the same problem. anyone could solve it?

**Comment by tesguru** [comment]
Have you guys solve it
Currently facing this issue on laravel 11

**Reply by masumcity** [reply]
same

**Reply by tesguru** [reply]
I finally solve the error!!! If your domain isn't running on HTTPS, it won't work. Set up HTTPS on your localhost or create a virtual host that supports HTTPS for the best results. You can download Laragon to create a virtual host for you. Just ensure that your virtual host is set to something like https://yourproject.dev.com to meet Google's TLD requirements.
